librz/include/rz_analysis.h

```diff
@@ -650,16 +655,36 @@ typedef struct rz_analysis_var_access_t {
 	ut8 type; ///< RzAnalysisVarAccessType bits
 } RzAnalysisVarAccess;
 
+struct rz_analysis_var_storage_t;
+
+typedef struct {
+	ut64 begin;
+	ut64 end;
+	struct rz_analysis_var_storage_t *storage;
+} RzAnalysisVarStorageListEntry;
+
 typedef enum {
+	RZ_ANALYSIS_VAR_STORAGE_INVALID = 0,
+	RZ_ANALYSIS_VAR_STORAGE_EMPTY,
+	RZ_ANALYSIS_VAR_STORAGE_DECODE_ERROR,
+	RZ_ANALYSIS_VAR_STORAGE_STACK,
 	RZ_ANALYSIS_VAR_STORAGE_REG,
-	RZ_ANALYSIS_VAR_STORAGE_STACK
+	RZ_ANALYSIS_VAR_STORAGE_REG_OFFSET,
+	RZ_ANALYSIS_VAR_STORAGE_CFA_OFFSET,
+	RZ_ANALYSIS_VAR_STORAGE_FB_OFFSET,
+	RZ_ANALYSIS_VAR_STORAGE_COMPOSITE,
+	RZ_ANALYSIS_VAR_STORAGE_LOCLIST,
+	RZ_ANALYSIS_VAR_STORAGE_DWARF_EVAL_WAITING,
+	RZ_ANALYSIS_VAR_STORAGE_END
 } RzAnalysisVarStorageType;
```

Comment:
I don't think we should extend the kinds of core variable storages. The problem with this is that it becomes insanely complicated to reason about the storage spaces (stack/regs). For example, there could be a `RZ_ANALYSIS_VAR_STORAGE_STACK` variable and a `RZ_ANALYSIS_VAR_STORAGE_REG_OFFSET` variable `rbp - 0x8` at the same time, which may overlap.
This would make more sophisticated analysis and decompiler integration like rz-ghidra practically almost impossible.

Ideally, all variables should only be reg or stack as before, with the stack values being precalculated, so something like `rbp - 0x8` from dwarf becomes a stack variable at the correct offset. FB_OFFSET may already be almost a stack variable if I interpret it right, just with some offset applied. The detailed dwarf-like info could be saved next to the core variable storage as some kind of "metadata" or "origin" inside the RzAnalysisVariable for further use if necessary.
However I do acknowledge that calculating the stack offset beforehand may not always be possible. If we really can't find another way, we may want to have only one additional storage type `RZ_ANALYSIS_VAR_STORAGE_EVAL_PENDING` or similar that indicates that one has to look at the detailed origin info to evaluate it.

Reply:
@imbillow sorry for the big change at such a later stage. Could you please change the storage to have an additional `RZ_ANALYSIS_VAR_STORAGE_EVAL_PENDING` and use it for storage that can't be evaluated upon loading. And some additional information in the variable structure to allow access details of it. 
But the composite one should remain in my opinion, since the concept transcends just DWARF and really just a compiler putting variable in multiple registers.

Reply:
Yes. You're absolutely right.

Reply:
> @imbillow sorry for the big change at such late stage. Could you please change the storage to have an additional `RZ_ANALYSIS_VAR_STORAGE_EVAL_PENDING` and use it for storage that can't be evaluated upon loading. And some additional information in the variable structure to allow access details of it. 
> 
> But the composite (`RZ_ANALYSIS_VAR_STORAGE_COMPOSITE`) one should remain in my opinion, since the concept transcends just DWARF and really just a compiler putting variable in multiple registers.

I am fine with it, it's just that this pr might have to be considered delayed to 0.7.

Reply:
Good point. Lets postpone it to 0.7.0 then.